frontend/src/metabase/embedding/embedding-iframe-sdk-setup/components/EmbeddingControlCard.tsx

```diff
@@ -0,0 +1,137 @@
+import { useRef, useState } from "react";
+import { match } from "ts-pattern";
+import { t } from "ttag";
+
+import { useUpdateSettingsMutation } from "metabase/api";
+import { useToast } from "metabase/common/hooks";
+import { Button, Card, Group, HoverCard, Icon, Stack, Text } from "metabase/ui";
+import type { EnterpriseSettings } from "metabase-types/api";
+
+interface EmbeddingControlCardProps {
+  embeddingType: string;
+  isEnabled: boolean;
+  termsAccepted: boolean;
+  settingsToUpdate: Partial<EnterpriseSettings>;
+  errorMessage: string;
+}
+
+export const EmbeddingControlCard = ({
+  embeddingType,
+  isEnabled,
+  termsAccepted,
+  settingsToUpdate,
+  errorMessage,
+}: EmbeddingControlCardProps) => {
+  const [updateSettings] = useUpdateSettingsMutation();
+  const [sendToast] = useToast();
+
+  const isAccepted = isEnabled && termsAccepted;
+
+  // Freeze the card visibility to show even when we enabled the embedding setting.
+  // This allows us to show the "Enabled" button state and not hide the card.
+  const [showCard] = useState(!isAccepted);
```

Comment:
nit: you can move this `isAccepted` useState to `initialDataRef` as it serves the same purpose of freezing the initial data